x-pack/plugins/security_solution/server/lib/detection_engine/signals/utils.ts

```diff
@@ -488,135 +416,151 @@ export const errorAggregator = (
   }, Object.create(null));
 };
 
+export const getRuleRangeTuples = ({
+  logger,
+  previousStartedAt,
+  from,
+  to,
+  interval,
+  maxSignals,
+  buildRuleMessage,
+}: {
+  logger: Logger;
+  previousStartedAt: Date | null | undefined;
+  from: string;
+  to: string;
+  interval: string;
+  maxSignals: number;
+  buildRuleMessage: BuildRuleMessage;
+}) => {
+  const tuples = [
+    getOriginalTuple({
+      ruleParamsFrom: from,
+      ruleParamsTo: to,
+      ruleParamsMaxSignals: maxSignals,
+      buildRuleMessage,
+    }),
+  ];
+  try {
+    const intervalDuration = parseInterval(interval);
```

Comment:
I tried to minimize the parsing of strings into dates and instead parse once and pass the parsed value around. I also tried to reduce the number of places we can return `null` or `undefined`, and instead do up-front checks so the rest of the code can rely on the values existing.

Here, for example, `parseInterval` can throw, in which case we know that the rest of code path here that relies on `intervalDuration` won't be able to complete so we skip straight to the `catch` rather than catching the error inside `parseInterval` and returning null.